annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastProcessor.java

```diff
@@ -19,112 +23,198 @@
                         "annotation added after first processing round. " +
                         "It may be caused by activity of other annotation processors.");
             }
-            fastTypesByName = collectFastTypes(roundEnv);
+            packageElementsMap = collectFastPackages(roundEnv);
+            HashMap<String, Element> fastTypesByName = collectFastTypes(roundEnv);
+            organizeInPackages(fastTypesByName);
             try {
-                buildCreatorClass(fastTypesByName);
+                buildCreatorClass(packageElementsMap);
             } catch (Exception e) {
-                processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, e.getMessage());
+                messager.printMessage(Diagnostic.Kind.ERROR, e.getMessage());
             }
             return true;
         }
-
         return false;
     }
 
+    private void organizeInPackages(HashMap<String, Element> fastTypesByName) {
+        for (Map.Entry<String, Element> entry : fastTypesByName.entrySet()) {
+            PackageElement enclosingPackage = elementUtils.getPackageOf(entry.getValue());
+            while (!packageElementsMap.containsKey(enclosingPackage)) {
+                enclosingPackage = getParent(enclosingPackage);
+            }
+            packageElementsMap.get(enclosingPackage).put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    private PackageElement getParent(Element enclosing) {
+        String currentPackage = enclosing.toString();
+        int lastDotIndex = currentPackage.lastIndexOf('.');
+        if (lastDotIndex  == -1) {
+            return elementUtils.getPackageElement("");
+        }
+        return elementUtils.getPackageElement(currentPackage.substring(0, lastDotIndex));
+    }
+
+    private HashMap<PackageElement, HashMap<String, Element>> collectFastPackages(RoundEnvironment roundEnv) {
+        HashMap<PackageElement, HashMap<String, Element>> packageElementMap = new HashMap<>();
+        PackageElement topPackage = elementUtils.getPackageElement("");
+        packageElementMap.put(topPackage, new HashMap<>());
+        Set<? extends Element> packageSet = roundEnv.getElementsAnnotatedWith(fastPackageElement);
+        for (Element element : packageSet) {
+            String packageName = element.getAnnotation(FastPackage.class).name();
+            if (packageName.isEmpty()) {
+                PackageElement packageElement = elementUtils.getPackageOf(element);
+                packageElementMap.put(packageElement, new HashMap<>());
+            } else {
+                try {
+                    PackageElement packageElement = elementUtils.getPackageElement(packageName);
+                    packageElementMap.put(packageElement, new HashMap<>());
+                } catch (Exception e) {
+                    messager.printMessage(
+                        Diagnostic.Kind.ERROR,
+                        String.format("Package for FastPackage name = %s doesn't exist", packageName),
```

Comment:
Based on this we really have to change the code (see previous comment). Can we use just store a name as a string instead of PackageElement?